src/hotspot/share/classfile/classLoader.cpp

```diff
@@ -1426,11 +1474,11 @@ jlong ClassLoader::classloader_time_ms() {
 }
 
 jlong ClassLoader::class_init_count() {
-  return UsePerfData ? _perf_classes_inited->get_value() : -1;
+  return (UsePerfData) ? _perf_classes_inited->get_value() : -1;
```

Comment:
No need to add brackets here

Reply:
Surely this needs to be checking `ProfileClassLinkage`, which in turn should be false if `UsePerfData` is false.

Reply:
If `UsePerfData` is set to false, `ProfileClassLinkage` is set to false in arguments.cpp:

```
3761   if (ProfileClassLinkage && !UsePerfData) {
3762     if (!FLAG_IS_DEFAULT(ProfileClassLinkage)) {
3763        warning("Disabling ProfileClassLinkage since UsePerfData is turned off.");
3764        FLAG_SET_DEFAULT(ProfileClassLinkage, false);
3765      }
3766   }
```

I will remove the extra parentheses.

Reply:
Yes but if  `UsePerfData` is true it doesn't mean `ProfileClassLinkage` is true.

Reply:
The `_perf_classes_inited` counter is a pre-existing counter only depends on `UsePerfData`. The `ProfileClassLinkage` should not have any effect on the counter.